src/app/service.rs

```diff
@@ -95,7 +95,7 @@ impl<T: Send + Sync> Service<http::Request<Incoming>> for AppService<T> {
         // segment.
         for (route, param) in self.app.router.traverse(parts.uri.path()) {
             // Add the matched route's middleware to the call stack.
-            next.extend(route.map(Arc::clone));
+            next.extend(route.cloned());
```

Comment:
Using `cloned()` eliminates the need to carry around a function pointer to `Arc::clone`.